WordPressKit/WordPressOrgXMLRPCApi.swift

```diff
@@ -451,3 +506,109 @@ extension WordPressOrgXMLRPCApiError: LocalizedError {
+                    // when XML-RPC is disabled for authenticated calls (e.g. xmlrpc_enabled is false on WP.org),
+                    // it will return a valid fault payload with a non-200
+                    return .failure(.endpointError(.init(response: response, code: decoder.faultCode(), message: decoder.faultString())))
+                } else {
+                    return .failure(.unacceptableStatusCode(response: response.response, body: response.body))
+                }
+            }
+
+            if ["application/xml", "text/xml"].filter({ (type) -> Bool in return contentType.hasPrefix(type)}).count == 0 {
+                return .failure(.unparsableResponse(response: response.response, body: response.body, underlyingError: WordPressOrgXMLRPCApiError.unknown))
+            }
+
+            guard let decoder = WPXMLRPCDecoder(data: response.body) else {
+                return .failure(.unparsableResponse(response: response.response, body: response.body))
+            }
+
+            guard !decoder.isFault() else {
+                return .failure(.endpointError(.init(response: response, code: decoder.faultCode(), message: decoder.faultString())))
+            }
+
+            if let decoderError = decoder.error() {
+                return .failure(.unparsableResponse(response: response.response, body: response.body, underlyingError: decoderError))
+            }
+
+            guard let responseXML = decoder.object() else {
+                return .failure(.unparsableResponse(response: response.response, body: response.body))
+            }
+
+            return .success(HTTPAPIResponse(response: response.response, body: responseXML as AnyObject))
+        }
+    }
+
+}
+
+private extension WordPressAPIError where EndpointError == WordPressOrgXMLRPCApiFault {
+
+    /// Convert to NSError for backwards compatiblity.
+    ///
+    /// Some Objective-C code in the WordPress app checks domain of the errors returned by `WordPressOrgXMLRPCApi`,
+    /// which can be WordPressOrgXMLRPCApiError or WPXMLRPCFaultErrorDomain.
+    ///
+    /// Swift code should avoid dealing with NSError instances. Instead, they should use the strongly typed
+    /// `WordPressAPIError<WordPressOrgXMLRPCApiFault>`.
+    func asNSError() -> NSError {
+        let error: NSError
+        let data: Data?
+        let statusCode: Int?
+        switch self {
+        case let .requestEncodingFailure(underlyingError):
+            error = underlyingError as NSError
+            data = nil
+            statusCode = nil
+        case let .connection(urlError):
+            error = urlError as NSError
+            data = nil
+            statusCode = nil
+        case let .endpointError(fault):
+            error = NSError(domain: WPXMLRPCFaultErrorDomain, code: fault.code ?? 0, userInfo: [NSLocalizedDescriptionKey: fault.message].compactMapValues { $0 })
+            data = fault.response.body
+            statusCode = nil
```

Comment:
Would this work?

```suggestion
            statusCode = fault.response.statusCode
```

Reply:
> Depending on what you mean by "work" 😄 .

Ahah. Good point @crazytonyli . Makes sense to be cautious 👍 

Besides, ideally the HTTP level details should remain at the networking library level, and the app should only need to check the typed errors it gets, without needing to look into HTTP error codes.

Reply:
> the app should only need to check the typed errors it gets, without needing to look into HTTP error codes.

Unfortunately that is not the case at the moment ([here is an example](https://github.com/search?q=%28repo%3Awordpress-mobile%2FWordPress-iOS+OR+repo%3Awordpress-mobile%2FWordPressAuthenticator-iOS%29+%22%3D%3D+403%22&type=code)), which is one of the reasons that I'd like to be cautious.